### DISCUSSION TITLE
how to regex a substr in duckdb?

### DISCUSSION THREAD

**Discussion by wonb168** [duckdb/duckdb#6264] [General]
### What happens?

I want get columns part from create table sql such as
```
create table test ( a int, b text) distribute by (a)
```
I want get `a int, b text`, In postgres it can:
```
select substring('create table test (a int, b text) distributed by (a)' from '\((.*)\) distributed')
```
but how to do this in duckdb?

### To Reproduce

same as above

### OS:

windows

### DuckDB Version:

0.7.0

### DuckDB Client:

cli

### Full Name:

wang cz

### Affiliation:

Linezone

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by wonb168** [comment]
```
select regexp_replace(regexp_replace(sql,'create table .*?\(',''),'\)\s+distributed','') cols from t
```
Now, I used regexp_replace twice to get the result, if duckdb support regexp_substr will be better!!!

**Reply by wonb168** [reply]
and, I found regexp_extract can do this, but can't support multi lines.

**Reply by hawkfish** [reply]
You can specify options as the last argument to `regexp_extract` as of issue #6882. Currently supported options are `cimnpsg`

**Reply by hawkfish** [reply]
Looks like the documentation is a bit lacking here.
